etc/requirements_dev.in

```diff
@@ -0,0 +1,23 @@
+# Testing
+coverage>=4.0.3
+nose>=1.3.7
+nose-parameterized>=0.5.0
+nose-ignore-docstring>=0.2
+nose-timer>=0.5.0
+
+mock>=2.0.0
+
+# Temp Directories for testing
+testfixtures>=4.1.2
+
+# Linting
+flake8>=3.3.0
+
+# Algo examples
```

Comment:
Not a blocker, but it seems unfortunate to have matplotlib in here. The rest of the dev requirements are pretty lightweight, but mpl requires a bunch of external dependencies. We might want to consider just making mpl fully optional, since none of our tests actually use it.